### PR TITLE
NFT components various improvements

### PR DESCRIPTION
**Pull request by jagodarybacka**
Resolves https://github.com/tallycash/extension/issues/2600
Resolves https://github.com/tallycash/extension/issues/2598
Resolves https://github.com/tallycash/extension/issues/2599
Resolves https://github.com/tallycash/extension/issues/2602

This PR includes many small improvements to the NFTs list components.

### Changes

- add NFT lists to NFT page
- if NFT has no name use placeholder "no title"
- if NFT has no preview image use placeholder (there is new placeholder design, will be added later)
image
- trim NFT description to 180 chars so about ~4 lines of text
- show creator address and # of items in collection
- show relevant marketplaces for each NFT on the preview
image
- hide "Send" button behind a feature flag
- make badges NFTs with blurred background (known bug: Galxe NFTs are not in the Badges tab, will fix later) 
image
image

Latest build: [extension-builds-2695](https://github.com/tallycash/extension/suites/9861777965/artifacts/475647527) (as of Wed, 14 Dec 2022 12:14:08 GMT).

### CODE REVIEW

File: ui/components/NFTS_update/NFTPreview.tsx [hunk 1]
```diff
@@ -81,49 +112,54 @@ export default function NFTPreview(props: NFTWithCollection): ReactElement {
...
-          <SharedButton
-            type="tertiary"
-            size="small"
-            iconSmall="send"
-            iconPosition="left"
-          >
-            {t("preview.send")}
-          </SharedButton>
+          <h1 className="ellipsis_multiline">
+            {name?.length ? name : "No title"}
```
Comment (kkosiorowska) [inline ui/components/NFTS_update/NFTPreview.tsx:116]:
```suggestion
            {name ||  "No title"}
```

File: ui/components/NFTS_update/NFTPreview.tsx [hunk 2]
```diff
@@ -81,49 +112,54 @@ export default function NFTPreview(props: NFTWithCollection): ReactElement {
...
+          <div className="preview_section_row preview_markets">
+            {marketsList.map(({ url, title, color, icon, getNFTLink }) => (
+              <ExploreMarketLink
+                type="button"
+                key={url}
+                url={getNFTLink(nft)}
+                {...{ title, color, icon }}
+              />
+            ))}
           </div>
```
Comment (kkosiorowska) [inline ui/components/NFTS_update/NFTPreview.tsx:141]: :heart:

File: ui/components/NFTS_update/NFTPreview.tsx [hunk 3]
```diff
@@ -81,49 +112,54 @@ export default function NFTPreview(props: NFTWithCollection): ReactElement {
...
               {t("preview.itemsCount")}
             </div>
-            <p>TODO</p>
+            <p>{totalNftCount ?? "-"}</p>
           </div>
           <div className="preview_section_column align_right">
             <div className="preview_section_header">{t("preview.creator")}</div>
-            <p>TODO</p>
+            <p>
+              {contract?.startsWith("0x") ? truncateAddress(contract) : "-"}
```
Comment (kkosiorowska) [inline ui/components/NFTS_update/NFTPreview.tsx:161]: You can use here `isProbablyEVMAddress` function

File: ui/components/Overview/NFTsPortfolioOverview.tsx [hunk 4]
```diff
@@ -1,19 +1,25 @@
...
+  const badgeCount = useBackgroundSelector(selectNFTBadgesCount)
 
+  const { totalFloorPriceInUSD } = useTotalNFTsFloorPrice()
   return (
     <section>
       <header>
-        <h5 className="nft-count">NFT(30)</h5>
-        <span className="estimate">~$24,231,00</span>
+        <h5 className="nft-count">NFT({nftCount + badgeCount})</h5>
+        <span className="estimate">~${totalFloorPriceInUSD}</span>
```
Comment (kkosiorowska) [inline ui/components/Overview/NFTsPortfolioOverview.tsx:22]: We can use `selectMainCurrencySign` for $

**Review by kkosiorowska — COMMENTED** [review]
It looks great :fire: I left some comments.

File: ui/components/NFTS_update/ExploreMarketLink.tsx [hunk 5]
```diff
@@ -1,13 +1,87 @@
...
+  color: string
+  icon: string
+  getNFTLink: (nft: NFT) => string
+}
+
+const OPENSEA_CHAINS = {
+  1: "ethereum",
+  10: "optimism",
+  137: "matic",
+  42161: "arbitrum",
```
[the remark this answers is not in the document]
Comment (greg-nagy) [reply ui/components/NFTS_update/ExploreMarketLink.tsx:26]: This feels like something that should probably be in `constants/networks.ts` Similar to `CHAIN_ID_TO_0X_API_BASE`

File: ui/components/NFTS_update/ExploreMarketLink.tsx [hunk 6]
```diff
@@ -18,7 +92,9 @@ export default function ExploreMarketLink({
   return (
     <a className={type} href={url} rel="noreferrer" target="_blank">
       {title}
-      <img width="16" height="16" src={`images/${icon}`} alt={title} />
+      {!!icon && (
```
Comment (greg-nagy) [inline ui/components/NFTS_update/ExploreMarketLink.tsx:95]: ❤️

File: ui/hooks/nft-hooks.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (greg-nagy) [inline ui/hooks/nft-hooks.ts:1]: I have a beef with this rule 💢

**Review by greg-nagy — APPROVED** [review]
the changes are looking good and working well 

Approving but better to merge the rest of the PRs into this and then merge this to main 
(Gh is funky when base branch is merged)

**Review by greg-nagy — APPROVED** [review]
let's 🚀

File: background/services/nfts/index.ts [hunk 8]
```diff
@@ -94,6 +95,32 @@ export default class NFTsService extends BaseService<Events> {
...
+        } else if (updatedNFTs.some((nft) => nft.isBadge)) {
+          // update collection as a badges collection
+          updatedCollection = await this.db.updateCollectionData(
+            collectionID,
+            account,
+            {
+              hasBadges: true,
+            }
+          )
+        }
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply background/services/nfts/index.ts]: Hmm, I think we are not wrapping db actions in try-catch often. If db action fails it is not breaking frontend (if I'm correct, it didn't when I checked) but it is visible in the console of background script.

File: background/redux-slices/selectors/nftsSelectors_update.ts [hunk 9]
```diff
@@ -66,11 +68,22 @@ export const selectNFTBadgesCount = createSelector(
...
+export const selectTotalFloorPriceInETH = createSelector(
+  selectAllCollections,
+  (collections) => {
+    return collections.reduce((sum, collection) => {
+      if (
+        collection.floorPrice &&
+        ETH_SYMBOLS.includes(collection.floorPrice.tokenSymbol)
+      ) {
+        return sum + collection.floorPrice.value * (collection.nftCount ?? 0)
+      }
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply background/redux-slices/selectors/nftsSelectors_update.ts:99]: Added task https://github.com/tallycash/extension/issues/2775

File: ui/components/NFTS_update/ExploreMarketLink.tsx [hunk 10]
```diff
@@ -1,13 +1,87 @@
...
+    title: "OpenSea",
+    url: "https://opensea.io/",
+    color: "#409FFF",
+    icon: "opensea.png",
+    getNFTLink: (nft: NFT): string =>
+      `https://opensea.io/assets/${
+        OPENSEA_CHAINS[
+          parseInt(nft.network.chainID, 10) as keyof typeof OPENSEA_CHAINS
+        ]
+      }/${nft.contract}/${nft.tokenId}`,
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply ui/components/NFTS_update/ExploreMarketLink.tsx:45]: hmm, idk as long as this is the only place we use it I think it's fine for now

**Pull request merged**